jdaviz/configs/imviz/plugins/viewers.py

```diff
@@ -196,11 +196,14 @@ def on_limits_change(self, *args):
             return
         self.set_compass(self.state.layers[i].layer)
 
-    def set_compass(self, image):
+    def set_compass(self, image, only_if_open=False):
         """Update the Compass plugin with info from the given image Data object."""
         if self.compass is None:  # Maybe another viewer has it
             return
 
+        if only_if_open and not self.compass.plugin_opened:
+            return
```

Comment:
When will the code ever enter here (not having already returned from the `self.compas is None`)?  I added a print statement and can't seem to find a scenario where its needed.

Reply:
The `@observe('plugin_opened')` on [_compass_with_new_viewer](https://github.com/spacetelescope/jdaviz/pull/1153/files#diff-d00ec66d428dae92765a78d062c61800d5c32561d2296f7f08b5d2058deca69eL41-R41) should handle re-drawing the compass as soon as the plugin is reopened (since it makes a call to `viewer.on_limits_change`).  It will then continue to redraw live from any limits changes if the plugin remains opened.  And the `if self.compass is None: return` at the top of `set_compass` will stop updating the compass while the plugin is closed (as all viewers have been set to have `viewer.compass = None` when the plugin was closed.

I don't think any of the `only_if_open` logic here is necessary at all, but if you find a case where that wasn't behaving as expected, I can look into it and try to fix it.

Reply:
Yup, you are right. I removed the extra unnecessary code. Good catch!